### PR TITLE
fix(multi-origin): Fix some issues with errors

### PR DESCRIPTION
**Pull request by chrisbreiding**
### User facing changelog

N/A

### Additional details

Fixes a couple issues with errors:

- When sending `run:origin:fn` errors, the userInvocationStack was not applied, so a code frame was not displayed
- When there was an uncaught error in the secondary after cy.origin finished, the error would get double-wrapped with `The following error originated from your test code...`

### PR Tasks

- [x] Have tests been added/updated?
- N/A Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- N/A Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- N/A Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- N/A Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/cypress/integration/e2e/multi-domain/multi_domain_spec.ts [hunk 1]
```diff
@@ -152,16 +152,15 @@ describe('cy.origin', () => {
     })
 
     describe('errors', () => {
-      // TODO: Proper stack trace printing still needs to be addressed here
-      // with a cy-in-cy test
-      // https://github.com/cypress-io/cypress/issues/20973
```
Comment (chrisbreiding) [inline packages/driver/cypress/integration/e2e/multi-domain/multi_domain_spec.ts:157]: Testing the codeFrame works well enough for this.

File: packages/driver/src/cy/multi-domain/index.ts [hunk 2]
```diff
@@ -189,13 +192,7 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
         // to stack up listeners on it after it's originally bound
         if (!communicator.listeners('uncaught:error').length) {
           communicator.once('uncaught:error', ({ err }) => {
-            if (err?.name === 'CypressError') {
-              // This is a Cypress error thrown from the secondary origin after the command queue has finished, do not wrap it as a spec or app error.
-              cy.fail(err, { async: true })
-            } else {
-              // @ts-ignore
-              Cypress.runner.onSpecError('error')({ error: err })
-            }
```
Comment (chrisbreiding) [inline packages/driver/src/cy/multi-domain/index.ts:198]: As far I can tell, calling `onSpecError` isn't necessary here and it just causes errors to get double-wrapped in certain circumstances. Without calling `onSpecError`, the errors still get wrapped properly (and all the tests pass with this change).

**Review by mschile — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]

File: packages/driver/src/cy/multi-domain/index.ts (hunk 2 above)
Comment (AtofStryker) [reply packages/driver/src/cy/multi-domain/index.ts:198]: @mjhenkes and I spent some time on this, but I can't remember why 😕 .

File: packages/driver/src/cy/multi-domain/index.ts [hunk 3]
```diff
@@ -164,7 +167,7 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
               })
 
               wrappedErr.name = err.name
-              wrappedErr.stack = err.stack
+              wrappedErr.stack = $stackUtils.replacedStack(wrappedErr, err.stack)
```
Comment (AtofStryker) [inline packages/driver/src/cy/multi-domain/index.ts:170]: so the error is being reified, but we need the stack to be updated in the new wrapped error?

**Review by AtofStryker — APPROVED** [review]

File: packages/driver/src/cy/multi-domain/index.ts (hunk 3 above)
Comment (chrisbreiding) [reply packages/driver/src/cy/multi-domain/index.ts:170]: Right, we want the message from`ran_origin_fn_reference_error`, but the stack of the original error.

We should probably update `$errUtils.throwErrByPath` to optionally accept a stack and name, so this manual massaging isn't necessary. Maybe in another PR.

**Pull request merged**